src/matcher.ts

```diff
@@ -6,36 +6,35 @@ import { Location, Matcher, Params, Search } from "./types";
 // Kudos to https://reach.tech/router/ranking
 const extractFromPathname = (pathname: string) => {
   const parts = pathname.split("/").filter(isNonEmpty);
-  const segments: Matcher["segments"] = [];
+  const path: Matcher["path"] = [];
 
   let ranking = parts.length > 0 ? parts.length * 4 : 5;
 
   for (const part of parts) {
     const param = isParam(part);
-    const name = param ? part.substring(1) : encodeURIComponent(part);
-
     ranking += param ? 2 : 3;
-    segments.push({ name, param });
+    path.push(param ? { name: part.substring(1) } : encodeURIComponent(part));
   }
 
-  return { ranking, segments };
+  return { ranking, path };
 };
 
 export const getMatcher = (name: string, route: string): Matcher => {
   if (route.endsWith("*")) {
     const { pathname = "/" } = parsePath(route.slice(0, -1));
-    const { ranking, segments } = extractFromPathname(pathname);
+    const { ranking, path } = extractFromPathname(pathname);
 
     return {
       finite: false,
       name,
       ranking: ranking - 1, // penality due to wildcard
-      segments,
+      path,
       search: {},
+      hash: undefined,
```

Comment:
Wouldn't it be interesting to have a `hash` with the pathname stripped from the `*`? (not sure it would __really__ be interesting, just a wild guess)

Reply:
It's planned in the "merging routes" feature release.

```tsx
createGroup("Repositories", "/users/:userId/repositories?:foo", {
  Area: "/*",
  Detail: "/:repositoryId",
});
```

Soon, by doing this, you'll be able to get get `foo` from params for all the routes in the group, and will have the possibility to pass `foo` for each route creation.

Currently adding this feature in this form:

```tsx
createGroup("Repositories", "/users/:userId/repositories", {
  RepositoriesArea: "/*?:foo",
  RepositoriesDetail: "/:repositoryId",
});
```

Would be weird, since you get `foo` from the `RepositoriesArea` matching, but cannot create an url with `foo` in it except if you add it to all the routes in group:

```tsx
createGroup("Repositories", "/users/:userId/repositories", {
  RepositoriesArea: "/*?:foo",
  RepositoriesDetail: "/:repositoryId?:foo",
});
```